src/panoramic/cli/command.py

```diff
@@ -251,3 +257,79 @@ def push(yes: bool = False, target_dataset: Optional[str] = None, diff: bool = F
             except Exception as e:
                 bar.write(f'Error: Failed to execute action {action.description}:\n  {str(e)}')
         bar.write(f'Updated {bar.total} models')
+
+
+def detect_joins(target_dataset: Optional[str] = None, diff: bool = False, overwrite: bool = False, yes: bool = False):
+    company_slug = get_company_slug()
+    echo_info('Loading local state...')
+    local_state = get_local_state(target_dataset=target_dataset)
+
+    if local_state.is_empty:
+        echo_info('No datasets to detect joins on')
+
+    join_detector = JoinDetector(company_slug=company_slug)
+    join_detector.fetch_token()
+
+    models_by_virtual_data_source: Dict[str, Dict[str, PanoModel]] = defaultdict(dict)
+    for model in local_state.models:
+        # Prepare a mapping for a quick access when reconciling necessary changes later
+        models_by_virtual_data_source[model.virtual_data_source][model.model_name] = model
+
+    actions_list = ActionList(actions=[])
+
+    with tqdm(list(local_state.data_sources)) as bar:
+        for dataset in bar:
+            try:
+                echo_info(f'Detecting joins for dataset {dataset.dataset_slug}')
+                joins_by_model = join_detector.detect(dataset.dataset_slug)
+
+                for model_name, joins in joins_by_model.items():
+                    if not joins:
+                        bar.write(f'No joins detected for {model_name} under dataset {dataset.dataset_slug}')
+                        continue
+
+                    bar.write(f'Detected {len(joins)} joins for {model_name} under dataset {dataset.dataset_slug}')
+
+                    detected_join_objects = [PanoModelJoin.from_dict(join_dict) for join_dict in joins]
+                    current_model = models_by_virtual_data_source[dataset.dataset_slug][model_name]
+                    desired_model = deepcopy(current_model)
+
+                    if overwrite:
+                        desired_model.joins = detected_join_objects
+                    else:
+                        for detected_join in detected_join_objects:
+                            # Only append joins that are not already defined
+                            if detected_join not in current_model.joins:
+                                desired_model.joins.append(detected_join)
+
+                    actions_list.actions.append(Action(current=current_model, desired=desired_model))
+
+            except JoinException as join_exception:
+                bar.write(str(join_exception))
+            except Exception as e:
+                bar.write(f'An unexpected error occured when detecting joins for {dataset.dataset_slug}.')
+                bar.write(str(e))
```

Comment:
Can we change the exception handling to be the same as what scan does? https://github.com/panoramichq/panoramic-cli/pull/4/files#diff-7cd4305f463e8a73e45fc3f203973a99R167-R172

Reply:
I.e. add `Error: ` prefix and only log the underlying exception to the logger

Reply:
Fixed in 742d698.